src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs

```diff
@@ -64,6 +65,11 @@ internal void Load(IDictionary envVariables)
                         prefix = MySqlServerPrefix;
                         provider = "MySql.Data.MySqlClient";
                     }
+                    else if (key.StartsWith(PostgreSqlServerPrefix, StringComparison.OrdinalIgnoreCase))
+                    {
+                        prefix = PostgreSqlServerPrefix;
+                        provider = "Npgsql";
```

Comment:
@roji Does this look right to you?

Reply:
Sure, I guess (don't have much context here)... Though note that Npgsql isn't the only PG provider out there, so equating PostgreSQL with Npgsql could be problematic.

The same problem seems actually more acute: for SQL Server, Microsoft.Data.SqlClient is the modern thing to be used rather than System.Data.SqlClient. And for MySQL there's [MySqlConnector](https://github.com/mysql-net/MySqlConnector), which is frequently better/faster.

Do we have to make a provider choice here, i.e. could we put "Postgres" instead of "Npgsql"? The application would then get the connection string for PG and pass it to Npgsql, assuming it's using that?